src/Fetch/Message.php

```diff
@@ -670,7 +670,7 @@ protected function processAddressObject($addresses)
         $outputAddresses = array();
         if (is_array($addresses))
             foreach ($addresses as $address) {
-                if (property_exists($address, 'mailbox') && $address->mailbox != 'undisclosed-recipients') {
+                if (property_exists($address, 'mailbox') && !empty($address->host)) {
```

Comment:
I'm not sure if this will work as expected. I would find this if better:
```php 
if (property_exists($address, 'mailbox') && !empty($address->host) && property_exists($address, 'host')) {

}
```
But perhaps it would be good that mbohlaender would also review this!